--- ./tools/clang/lib/Driver/ToolChains.cpp.orig	2016-12-24 18:28:26.000000000 -0800
+++ ./tools/clang/lib/Driver/ToolChains.cpp	2016-12-24 18:29:10.000000000 -0800
@@ -898,8 +898,8 @@
     }
   }
 
-  // Default to use libc++ on OS X 10.9+ and iOS 7+.
-  if (((isTargetMacOS() && !isMacosxVersionLT(10, 9)) ||
+  // Default to use libc++ on OS X 10.4+ and iOS 7+.
+  if (((isTargetMacOS() && !isMacosxVersionLT(10, 4)) ||
        (isTargetIOSBased() && !isIPhoneOSVersionLT(7, 0))) &&
       !Args.getLastArg(options::OPT_stdlib_EQ))
     DAL->AddJoinedArg(nullptr, Opts.getOption(options::OPT_stdlib_EQ),
