diff --git modules/gui/macosx/intf.m modules/gui/macosx/intf.m
index 29d8dfe..d3bc081 100644
--- modules/gui/macosx/intf.m
+++ modules/gui/macosx/intf.m
@@ -705,7 +705,7 @@ static VLCMain *_o_sharedMainInstance = nil;
 
     o_vout_controller = [[VLCVoutWindowController alloc] init];
 
-    informInputChangedQueue = dispatch_queue_create("org.videolan.vlc.inputChangedQueue", DISPATCH_QUEUE_SERIAL);
+    informInputChangedQueue = dispatch_queue_create("org.videolan.vlc.inputChangedQueue", NULL);
 
     return _o_sharedMainInstance;
 }
diff --git modules/video_output/macosx.m modules/video_output/macosx.m
index 50ef29f..f9a115c 100644
--- modules/video_output/macosx.m
+++ modules/video_output/macosx.m
@@ -773,11 +773,6 @@ static void OpenglSwap (vlc_gl_t *gl)
     NSRect videoRect = [self bounds];
     BOOL b_inside = [self mouse: ml inRect: videoRect];
 
-    if (OSX_LION) {
-        ml = [self convertPointToBacking: ml];
-        videoRect = [self convertRectToBacking: videoRect];
-    }
-
     if (b_inside) {
         @synchronized (self) {
             if (vd) {
